pp/go/relabeler/head/head.go

```diff
@@ -164,20 +164,27 @@ type Head struct {
 	generation uint64
 	readOnly   bool
 
-	relabelersData             map[string]*RelabelerData
+	relabelersData map[string]*RelabelerData
+	rdMutex        sync.Mutex
+
 	dataStorages               []*DataStorage
 	wals                       []*ShardWal
 	lsses                      []*LSS
 	stageInputRelabeling       []chan *TaskInputRelabeling
 	stageAppendRelabelerSeries []chan *TaskAppendRelabelerSeries
+	stageUpdateRelabelers      []chan *TaskUpdateRelabelerState
 	genericTaskCh              []chan *GenericTask
+	genericReadTaskCh          []chan *GenericReadTask
 	numberOfShards             uint16
 	// stat
 	appendedSegmentCount prometheus.Counter
 	memoryInUse          *prometheus.GaugeVec
 	series               prometheus.Gauge
 	queried              *prometheus.GaugeVec
 	walSize              *prometheus.GaugeVec
+	queueInputRelabeling *prometheus.GaugeVec
+	queueRead            *prometheus.GaugeVec
+	queueGeneric         *prometheus.GaugeVec
```

Comment:
Using gauge for channel length is not representable. It is better use pair of counters: append, read. But of course instrumenting with counters is more difficult and verbose. Let's plan refactoring this metrics in future